src/tests/functional/frame_tests.ts

```diff
@@ -397,9 +400,11 @@ test("test evaluates frame script elements on each render", async ({ page }) =>
   assert.equal(await frameScriptEvaluationCount(page), undefined)
 
   await page.click("#body-script-link")
+  await nextEventOnTarget(page, "body-script", "turbo:frame-load")
   assert.equal(await frameScriptEvaluationCount(page), 1)
 
   await page.click("#body-script-link")
+  await nextEventOnTarget(page, "body-script", "turbo:frame-load")
   assert.equal(await frameScriptEvaluationCount(page), 2)
 })
```

Comment:
This test was correct, but race conditions meant that sometimes it tries to evaluate the count too soon. So changed to wait for the load to finish before each check.